docs/articles/documentation/extending-testcafe/custom-reporter/index.md

```diff
@@ -0,0 +1,232 @@
+
+reportTestDone (/* name, errs, durationMs, unstable, screenshotPath */) {
+    throw new Error('Not implemented');
+},
+
+reportTaskDone (/* endTime, passed */) {
+    throw new Error('Not implemented');
+}
+```
+
+These methods should output the desired information at certain moments during test run.
+
+* **reportTaskStart** - fires when a test task is started, that is at the beginning of testing.
+
+* **reportFixtureStart** - fires each time a fixture is started.
+
+* **reportTestDone** - fires each time a test is finished.
+
+* **reportTaskDone** - fires when the entire task is finished, that is at the end of testing.
+
+All the required data is provided to these methods through their parameters.
+
+To output information, use [helper methods](helper-methods.md) and libraries.
+TestCafe will mixin the helper methods to the reporter, so that you can access the helper methods by using `this`.
+
+In the `src/index.js` file, you can also enable or disable coloring a reporter output by using the `noColors` property. To color the output, use the [chalk](helper-methods.md#chalk) helper method.
+
+**Example**
+
+The following example demonstrates how you can implement four main methods,
+so that a generated report contains information about user agents used for testing,
+a result of individual test execution, summary information about passed/failed tests,
+and overall duration of tests.
+
+```js
+reportTaskStart (startTime, userAgents, testCount) {
+    this.startTime = startTime;
+    this.testCount = testCount;
+
+    this.write(`Running tests in: ${userAgents}`)
+        .newline()
+        .newline();
+},
+
+reportFixtureStart (name) {
+    this.currentFixtureName = name;
+},
+
+reportTestDone (name, errs) {
+    const hasErr    = !!errs.length;
+    const result    = hasErr ? `passed` : `failed`;
+
+    name = `${this.currentFixtureName} - ${name}`;
+
+    const title = `${result} ${name}`;
+
+    this.write(title)
+        .newline();
+},
+
+reportTaskDone (endTime, passed) {
+    const durationMs  = endTime - this.startTime;
+    const durationStr = this.moment.duration(durationMs).format('h[h] mm[m] ss[s]');
+    let footer      = passed === this.testCount ?
+                      `${this.testCount} passed` :
+                      `${this.testCount - passed}/${this.testCount} failed`;
+
+    footer += ` (Duration: ${durationStr})`;
+
+    this.write(footer)
+        .newline();
+}
+```
+
+This is a sample report generated by this reporter:
+
+```text
+Running tests in: Chrome,Firefox
+
+failed fixture1 - fixture1test1
+
+passed fixture1 - fixture1test2
+
+failed fixture1 - fixture1test3
+
+failed fixture2 - fixture2test1
+
+failed fixture2 - fixture2test2
+
+passed fixture3 - fixture3test1
+
+2/6 failed (Duration: 15m 25s)
+```
+
+## Building Reporter
+
+You can build a reporter project by using the `build` Gulp task.
+
+```bash
+gulp build
+```
+
+## Testing Reporter
+
+To test your reporter works well, you can use the `test` Gulp task.
```

Comment:
I've checked this sentence with Elsa Castanjo. She suggests the following variant: "To test your reporter operates well, you can use the `test` Gulp task."